crates/bevy_asset/src/id.rs

```diff
@@ -13,7 +14,7 @@ use std::{
 /// For an identifier tied to the lifetime of an asset, see [`Handle`].
 ///
 /// For an "untyped" / "generic-less" id, see [`UntypedAssetId`].
-#[derive(Reflect)]
+#[derive(Component, Reflect)]
```

Comment:
This shouldn't be a `Component` though. I'd prefer if it was newtyped

```rust
#[derive(Component)]
struct RenderHandleRef<A: Asset>(AssetId<A>);
```

making this a component may be misleading for users, they might accidentally insert an `AssetId` when they mean to insert a `Handle`

Reply:
I had discussed this with Cart and we thought it should become a component. The only reason I needed to change it was because ExtractComponent is implemented for `Handle<T>` and is supposed to extract to a `Handle<T>` at the moment, and we rather thought it should extract to an `AssetId<T>`.

Reply:
I still think it should be a newtype. I can see the the disadvantages but not the advantages (apart from expediency,  I'd be glad to open a PR after this one is merged if that's the only problem). Maybe there is an advantage I'm missing. Could you link me to the conversation?

Reply:
I can't as it was in DMs on Discord. On the batching PR, Cart mentioned that `AssetId<T>` should be used if a weak handle is desired, such as how I was using it there. It felt appropriate to also switch to `AssetId<T>` as part of this PR.

I am happy to wrap `AssetId<T>` in a newtype, but I don't understand why I would do it. I hear that you think a user might insert an `AssetId<T>` when they should have used a `Handle<T>` but I don't understand why wrapping it in a newtype would make much difference to that.

Reply:
You mentioned on Discord that it's easy for someone to see Handle<T> and AssetId<T> are both Components and think they can use either, whereas a newtype wrapper around AssetId<T> gives a bit more friction. I decided to revert the change that made AssetId<T> a Component and revert the change to the ExtractComponent impl for Handle<T>. I think it's better taken in a separate PR. :) Thanks for noticing it!